modules/base/include/modules/base/algorithm/randomutils.h

```diff
@@ -261,39 +265,41 @@ std::shared_ptr<Volume> randomVolume(size3_t dims, Rand& randomNumberGenerator =
 }
 
 /**
- * Generate an Image with perlin noise, a cloud like noise using the sum of several white noise
- * images with different frequencies
- * @param dims Size of the output image
+ * Generate a Layer with perlin noise, a cloud like noise using the sum of several white noise
+ * layers with different frequencies
+ * @param dims Size of the output layer
  * @param persistence controls the amplitude used in the different frequencies
  * @param startLevel controls the min level used. The level is determining the frequency to use
- * in each white noise image as 2^level
+ * in each white noise layer as 2^level
  * @param endLevel controlsthe max level used.
  * @param randomNumberGenerator the Random number generator to use, defaults to the Mersenne Twister
  * engine (std::mt19937)
  */
 template <typename Rand = std::mt19937>
-std::shared_ptr<Image> perlinNoise(size2_t dims, float persistence, size_t startLevel,
+std::shared_ptr<Layer> perlinNoise(size2_t dims, float persistence, size_t startLevel,
                                    size_t endLevel, Rand& randomNumberGenerator = Rand()) {
-    std::shared_ptr<Image> img = std::make_shared<Image>(dims, DataFloat32::get());
-    auto ram = static_cast<LayerRAMPrecision<float>*>(
-        img->getColorLayer()->getEditableRepresentation<LayerRAM>());
-
-    auto size = nextPow2(std::max(dims.x, dims.y));
-    std::vector<std::shared_ptr<Image>> levels;
+    const auto size = nextPow2(std::max(dims.x, dims.y));
+    std::vector<std::shared_ptr<Layer>> levels;
     std::vector<TemplateImageSampler<float, float>> samplers;
     auto currentSize = std::pow(2, startLevel);
     auto iterations = endLevel - startLevel + 1;
     float currentPersistance = 1;
     while (currentSize <= size && iterations--) {
-        size2_t imgsize{static_cast<size_t>(currentSize)};
+        size2_t layerSize{static_cast<size_t>(currentSize)};
         auto dist = std::uniform_real_distribution<float>(-currentPersistance, currentPersistance);
-        auto img1 = util::randomImage<float>(imgsize, randomNumberGenerator, dist);
-        samplers.push_back(TemplateImageSampler<float, float>(img1.get()));
-        levels.push_back(img1);
+        auto randomLayer = util::randomLayer<float>(layerSize, randomNumberGenerator, dist);
+        samplers.push_back(TemplateImageSampler<float, float>(randomLayer.get()));
+        levels.push_back(randomLayer);
         currentSize *= 2;
         currentPersistance *= persistence;
     }
 
+    std::shared_ptr<Layer> layer =
```

Comment:
create the LayerRAMPrecision first and add that to the layer...

Reply:
this might even just work with LayerRAMPrecision<float> no need for layers here?